File: Sorting/Ruby/InsertionSort.rb
def insertionSort(arr)
	for i in (1..arr.length-1) do
		temp = arr[i]
		j = i-1
		while (arr[j]> temp and j >= 0) do
			arr[j+1] = arr[j]
			j -= 1
		arr[j+1] = temp
		end # end while
	end # end for
	return arr
end # end func

File: Sorting/Java/InsertionSort.java

public class InsertionSort implements SortingAlgorithm {
	public void sort(int[]a){
		int temp; // Declare temp variable for swapping
		for (int i =1;i<a.length;i++){ // First pointer from the beginning
			for(int j=i; j>0;j--){ // Second pointer to scan from the end to i
				if (a[j]<a[j-1]){
					temp = a[j];
					a[j]=a[j-1];
					a[j-1] = temp;
				}
					
			}
		}
	
	}
	
} // End class
